micall/utils/consensus_aligner.py

```diff
@@ -953,6 +953,21 @@ def coord_concordance(self, half_window_size=10):
             concordance = sum(query_matches[pos - half_window_size:pos + half_window_size])
             normalized_concordance = concordance / (2 * half_window_size)
             concordance_list[pos] = normalized_concordance
+        # This fills the concordance everywhere except the positions within a half_window_size
+        # of the start and end of the consensus.
+        # We will do these next, with a window size decreasing from 2x half_window_size to half_window_size
+
+        if len(self.consensus) >= half_window_size:
+            # fill in the start
+            for pos in range(0, half_window_size):
+                concordance = sum(query_matches[0:pos + half_window_size])
+                normalized_concordance = concordance / (pos + half_window_size)
+                concordance_list[pos] = normalized_concordance
+            # fill in the end
+            for pos in range(len(self.consensus) - half_window_size + 1, len(self.consensus)):
+                concordance = sum(query_matches[pos-half_window_size:])
+                normalized_concordance = concordance / (len(self.consensus) - pos + half_window_size)
+                concordance_list[pos] = normalized_concordance
```

Comment:
Does it make sense to combine these three loops back into one, calculating `start` and `end` positions of the window for each iteration using `max(start, 0)` and `min(end, len(self.consensus))`? What happens with a very short consensus?

Reply:
Before, with a very short consensus, we did not compute any concordance. I like your suggestion, so I've now switched this to a simplified calculation where all three loops are folded into one and we average over the full window size if we can, and over whatever we have, if we cannot.